Rocket.Chat/Managers/Model/AuthSettingsManager.swift

```diff
@@ -26,7 +26,32 @@ final class AuthSettingsManager {
         }
     }
 
-    static func updatePublicSettings(serverVersion: Version? = nil, apiHost: URL? = nil, _ auth: Auth?, completion: @escaping MessageCompletionObject<AuthSettings>) {
+    static func persistPublicSettings(settings: AuthSettings, realm: Realm? = Realm.current, completion: (MessageCompletionObject<AuthSettings>)? = nil) {
```

Comment:
@rafaelks this is always called as a callback of an async operation, if you capture the context of Realm here you will never know if it is the same context that it was when the async operation started.